Newsletters: add 346 (2025-03-21) (#2228)
bitcoinops/bitcoinops.github.io: _posts/en/newsletters/2025-03-21-newsletter.md

    node or a third party.  This is mainly used for choosing initial
    feerates and fee bumping non-urgent transactions.

  - Exponential fee bumping, used when a deadline is approaching to
    ensure that problems with a node's mempool or its fee estimation
    don't prevent timely confirmation.  For example, Eclair uses exponential fee
    bumping when deadlines are within six blocks.

  Morehouse then describes how these two strategies are combined in
  LND's new sweeper system: "[HTLC][topic htlc] claims with matching
  deadlines [are aggregated] into a single [batched transaction][topic
  payment batching]. The budget for the batched transaction is
  calculated as the sum of the budgets for the individual HTLCs in the
  transaction. Based on the transaction budget and deadline, a fee
  function is computed that determines how much of the budget is spent
  as the deadline approaches.  By default, a linear fee function is used
  which starts at a low fee (determined by the minimum relay fee rate or
  an external estimator) and ends with the total budget being allocated
  to fees when the deadline is one block away."

>   He additionally describes how the new logic helps protected against

Reviewer: ```suggestion
  He additionally describes how the new logic helps protect against
```

  [replacement cycling][topic replacement cycling] attacks, concluding:
  "with LND’s default parameters an attacker must generally spend at
  least 20x the value of the HTLC to successfully carry out a
  replacement cycling attack."  He adds that the new system also
  improves LND's defense against [pinning attacks][topic transaction
  pinning].

  He concludes with a link-filled summary of several "LND-specific bug

>   and vulnerability fixes" made though the improved logic.  Abubakar

Reviewer: ```suggestion
  and vulnerability fixes" made through the improved logic.  Abubakar
```

  Sadiq Ismail [replied][ismail sweep] with some suggestions for how all

>   LN implementations (in addition to other software) can make more
>   effective use of Bitcoin Core's fee estimation.  Several other

Reviewer: ```suggestion
  LN implementations (in addition to other software) can more
  effectively use Bitcoin Core's fee estimation.  Several other
```

popular Bitcoin infrastructure software.

## News

- **Discussion of LND's dynamic feerate adjustment system:** Matt
  Morehouse [posted][morehouse sweep] to Delving Bitcoin a description
  of LND's recently-rewritten _sweeper_ system, which
  determines the feerates to use for onchain transactions (including
  [RBF][topic rbf] fee bumps).  He begins with a brief description
  of the safety critical aspects of fee management for an LN node, as
  well as the natural desire to avoid overpaying.  He then describes the
  two general strategies used by LND:

  - Querying external feerate estimators, such as a local Bitcoin Core
    node or a third party.  This is mainly used for choosing initial
    feerates and fee bumping non-urgent transactions.

  - Exponential fee bumping, used when a deadline is approaching to
    ensure that problems with a node's mempool or its [fee estimation][topic fee estimation]
    don't prevent timely confirmation.  For example, Eclair uses exponential fee
    bumping when deadlines are within six blocks.

  Morehouse then describes how these two strategies are combined in
  LND's new sweeper system: "[HTLC][topic htlc] claims with matching

>   deadlines [are aggregated] into a single [batched transaction][topic

Reviewer 2: > [are aggregated]

Is this a dead link?

Reviewer 3: I guess these are externally inserted words into quote.

Reviewer 2: I see. thanks!

  [available][second blog] on [signet][topic signet] with a faucet and demo
  store for testing.

- **Cove Bitcoin wallet announced:**
  [Cove Wallet][cove wallet github] is an open source Bitcoin mobile wallet based on
  BDK that supports technologies like PSBTs, [wallet labels][topic wallet labels],
  hardware signing devices, and more.

## Releases and release candidates

_New releases and release candidates for popular Bitcoin infrastructure
projects.  Please consider upgrading to new releases or helping to test
release candidates._

FIXME:harding to update or remove Thursday

## Notable code and documentation changes

_Notable recent changes in [Bitcoin Core][bitcoin core repo], [Core
Lightning][core lightning repo], [Eclair][eclair repo], [LDK][ldk repo],
[LND][lnd repo], [libsecp256k1][libsecp256k1 repo], [Hardware Wallet
Interface (HWI)][hwi repo], [Rust Bitcoin][rust bitcoin repo], [BTCPay
Server][btcpay server repo], [BDK][bdk repo], [Bitcoin Improvement
Proposals (BIPs)][bips repo], [Lightning BOLTs][bolts repo],
[Lightning BLIPs][blips repo], [Bitcoin Inquisition][bitcoin inquisition
repo], and [BINANAs][binana repo]._

> - [Bitcoin Core #31649][] removes all checkpoint logic, replacing it with the

Reviewer 4: "replacing it with" in this context sounds like this PR is replacing it, but really presync was a few years ago. Id suggest some slight rewording here @Gustavojfe

  of the safety critical aspects of fee management for an LN node, as
  well as the natural desire to avoid overpaying.  He then describes the
  two general strategies used by LND:

  - Querying external feerate estimators, such as a local Bitcoin Core
    node or a third party.  This is mainly used for choosing initial
    feerates and fee bumping non-urgent transactions.

  - Exponential fee bumping, used when a deadline is approaching to
    ensure that problems with a node's mempool or its [fee estimation][topic fee estimation]
    don't prevent timely confirmation.  For example, Eclair uses exponential fee
    bumping when deadlines are within six blocks.

  Morehouse then describes how these two strategies are combined in
  LND's new sweeper system: "[HTLC][topic htlc] claims with matching
  deadlines [are aggregated] into a single [batched transaction][topic
  payment batching]. The budget for the batched transaction is
  calculated as the sum of the budgets for the individual HTLCs in the
  transaction. Based on the transaction budget and deadline, a fee
  function is computed that determines how much of the budget is spent

>   as the deadline approaches.  By default, a linear fee function is used

Reviewer: ```suggestion
  as the deadline approaches.  By default, a linear fee function is used,
```

  hardware signing devices, and more.

## Releases and release candidates

_New releases and release candidates for popular Bitcoin infrastructure
projects.  Please consider upgrading to new releases or helping to test
release candidates._

- [Bitcoin Core 29.0rc2][] is a release candidate for the next major
  version of the network's predominate full node.

## Notable code and documentation changes

_Notable recent changes in [Bitcoin Core][bitcoin core repo], [Core
Lightning][core lightning repo], [Eclair][eclair repo], [LDK][ldk repo],
[LND][lnd repo], [libsecp256k1][libsecp256k1 repo], [Hardware Wallet
Interface (HWI)][hwi repo], [Rust Bitcoin][rust bitcoin repo], [BTCPay
Server][btcpay server repo], [BDK][bdk repo], [Bitcoin Improvement
Proposals (BIPs)][bips repo], [Lightning BOLTs][bolts repo],
[Lightning BLIPs][blips repo], [Bitcoin Inquisition][bitcoin inquisition
repo], and [BINANAs][binana repo]._

- [Bitcoin Core #31649][] removes all checkpoint logic, which is no longer
  necessary following the headers presync step implemented years ago (see
  Newsletter [#216][news216 presync]) that enables a node during Initial Block

>   Download (IBD) to determine if a chain of headers is valid by comparing its

Reviewer: ```suggestion
  Download (IBD) to determine if a chain of headers may be valid by comparing its
```
Maybe, because otherwise it sounds like that's the only validity check that's needed.

  total proof of work (PoW) to a predefined threshold `nMinimumChainWork`. Only
  chains with a total proof of work exceeding this value are considered valid
  and stored, effectively preventing memory DoS attacks from low-work headers.
  This eliminates the need for checkpoints, which were often seen as a
  centralized element.

- [Bitcoin Core #31283][] introduces a new `waitNext()` method to the
  `BlockTemplate` interface, which will only return a new template if the chain
  tip changes or the mempool fees increase above the `MAX_MONEY` threshold.
  Previously, miners would receive a new template with every request, resulting
  in unnecessary template generation. This change aligns with the [Stratum
  V2][topic pooled mining] protocol specification.

- [Eclair #3037][] enhances the `listoffers` command (See Newsletter
  [#345][news345 offers]) to return all relevant [offer][topic offers] data,

>   including the `createdAt` and `disabledAt` timestamps instead of just raw

Reviewer: ```suggestion
  including the `createdAt` and `disabledAt` timestamps, instead of just raw
```
